File: Easy/strangeNum/strangeNum.java
//run as strangeNum
import java.util.Scanner;
import java.lang.Math;

public class strangeNum{
	public static void main(String[] args){
		int T;
		long X,K,count;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=1000) : "T must be 1 <= T <= 1000";

		while(T>0){
			X = scan.nextLong();
			K = scan.nextLong();

			assert(X>=1 && X<=1000000000) : "X must be 1 <= X <= 10^9";
			assert(K>=1 && K<=1000000000) : "K must be 1 <= K <= 10^9";

			count = 0;

			while(X%2==0){
				count++;
				X /= 2;
			}

			for(long i=3;i<=Math.sqrt(X);i+=2){
				while(X%i==0){
					count++;
					X /= i;
				}
			}

			if(X>2){
				count++;
			}

			if(count>=K){
				System.out.println("1");
			}else{
				System.out.println("0");
			}

			T--;
		}
	}
}

File: Easy/strangeNum/strangeNum.c
#include <stdio.h>
#include <assert.h>
#include <math.h>

int main(void){
	int T;
	long long int X,K,count;

	scanf("%d",&T);
	assert(T>=1 && T<=1000);

	while(T>0){
		scanf("%lld %lld",&X,&K);

		assert(X>=1 && X<=1000000000);
		assert(K>=1 && K<=1000000000);

		count = 0;

		while(X%2==0){
			count++;
			X /= 2;
		}

		for(long long int i=3;i<=sqrt(X);i+=2){
			while(X%i==0){
				count++;
				X /= i;
			}
		}

		if(X>2){
			count++;
		}

		if(count>=K){
			printf("1\n");
		}else{
			printf("0\n");
		}

		T--;
	}
	
	return 0;
}